app/controllers/users_controller.rb

```diff
@@ -8,21 +8,60 @@ def show
   end
 
   def update
-    if @user.update(user_params)
-      redirect_to @user, notice: 'The user has been updated correctly.'
+    pass_params = password_params.to_h
+    exist_pass_arg = pass_params.any? { |k, v| !v.empty? and !v.nil? }
+    empty_pass_arg = pass_params.any? { |k, v| v.empty? or v.nil? }
+
+    if exist_pass_arg and empty_pass_arg
```

Comment:
this if/elseif contains to much code try to refactor this part, maybe moving logic to a private methods using move method and naming method patterns